test/value-table.test.ts

```diff
@@ -0,0 +1,47 @@
+import test from "ava";
+import { TableCell as TableCellInternal } from "@prismicio/types-internal/lib/customtypes";
+
+import { snapshotTwiceMacro } from "./__testutils__/snapshotTwiceMacro";
+
+import * as value from "../src/value";
+import { TableCell } from "../src/value/table";
+
+test("creates a mock Table field value", snapshotTwiceMacro, (t) =>
+	value.table({ seed: t.title }),
+);
+
+test("supports number seed", snapshotTwiceMacro, () =>
+	value.table({ seed: 1 }),
+);
+
+test("can be configured to return an empty value", (t) => {
+	const actual = value.table({
+		seed: t.title,
+		state: "empty",
+	});
+
+	t.is(actual, null);
+});
+
+test("provides a non empty Table head and body with rich text content", (t) => {
+	const actual = value.table({ seed: t.title });
+
+	t.is(actual.head?.rows.length, 1);
+	t.is(actual.head?.rows[0].cells.length, 3);
+	actual.head?.rows[0].cells.forEach((cell) => {
+		t.is(cell.type, "header");
+		t.truthy(cell.content.length);
+	});
+	t.is(actual.body.rows.length, 3);
+	actual.body.rows.forEach((row) => {
+		t.is(row.cells.length, 3);
+		row.cells.forEach((cell) => {
+			t.is(cell.type, "data");
+			t.truthy(cell.content.length);
+		});
+	});
+});
+
+test("ensures the correct table cell model is used", (t) => {
+	t.deepEqual(TableCell, TableCellInternal);
+});
```

Comment:
❓ #ask: Do we need this check if TypeScript already verifies it in the source?

Reply:
what we're doing this is to verify that the model value are exactly the same, i.e. we should not use a rich text node type in the content mocks that does not exist in the model value defined for types internal; so this is not about types, but about the actual value